src/build.py

```diff
@@ -492,61 +495,66 @@ def SyncPrebuiltClang(name, src_dir, git_repo):
   return ('chromium-clang', tools_clang)
 
 
-def SyncPrebuiltCMake(name, src_dir, git_repo):
-  if os.path.isdir(PREBUILT_CMAKE_DIR):
-    print 'Prebuilt CMake directory already exists'
-  else:
-    platform = 'Darwin' if sys.platform == 'darwin' else 'Linux'
-    filename = PREBUILT_CMAKE_ARCHIVE % platform
-    url = PREBUILT_CMAKE_URL + filename
-    Mkdir(PREBUILT_CMAKE_DIR)
-    try:
-      response = urllib2.urlopen(url)
-      data = response.read()
-      print 'Downloaded %s' % url
-      with tempfile.TemporaryFile() as f:
-        f.write(data)
-        f.seek(0)
-        # The tar file itself includes the 'cmake343' directory, so set the
-        # extract path to WORK_DIR to get the right path
-        tarfile.open(mode='r:gz', fileobj=f).extractall(path=WORK_DIR)
-        assert os.path.isfile(PREBUILT_CMAKE_BIN)
-      print 'Extracted CMake to %s' % PREBUILT_CMAKE_DIR
-    except urllib2.URLError as e:
-      print 'Error downloading %s: %s' % (url, e)
-      raise
-
-
-def SyncTarball(out_dir, name, version, url, tar):
+def SyncArchive(out_dir, name, version, url):
   if os.path.isdir(out_dir):
     print '%s directory already exists' % name
     return
   print 'Downloading %s %s from %s' % (name, version, url)
-  f = urllib2.urlopen(url)
-  print 'URL: %s' % f.geturl()
-  print 'Info: %s' % f.info()
-  with open(tar, 'wb') as out:
-    out.write(f.read())
-  proc.check_call(['tar', '-xvf', tar], cwd=WORK_DIR)
-  assert os.path.isdir(out_dir), 'Untar should produce %s' % out_dir
+  try:
+    f = urllib2.urlopen(url)
+    print 'URL: %s' % f.geturl()
+    print 'Info: %s' % f.info()
+    with tempfile.TemporaryFile()as t:
+      t.write(f.read())
+      t.seek(0)
+      print 'Extracting...'
+      if url.endswith('zip'):
+        with zipfile.ZipFile(t, 'r') as zip:
+          zip.extractall(path=WORK_DIR)
+      else:
+        tarfile.open(mode='r', fileobj=t).extractall(path=WORK_DIR)
+  except urllib2.URLError as e:
+    print 'Error downloading %s: %s' % (url, e)
+    raise
+
+
+def SyncPrebuiltCMake(name, src_dir, git_repo):
+  if os.path.isdir(PREBUILT_CMAKE_DIR):
+    print 'Prebuilt CMake directory already exists'
+  else:
+    platform = {'linux2': 'Linux',
```

Comment:
nit: Apparently some systems are `'linux3'`, and Python3.3 and on have `sys.platform` just return `'linux'`, so the most-correct check here is `sys.platform.startswith('linux')` which mucks up the convenience of using a dict somewhat.

This is probably fine until we abstract out all this platform-y stuff.